package/helper_scripts/check_for_default_sa.sh

```diff
@@ -8,23 +8,22 @@ handle_error() {
 
 trap 'handle_error' ERR
 
-count_sa=$(kubectl get serviceaccounts --all-namespaces -o json | jq -r '.items[] | select(.metadata.name=="default") | select((.automountServiceAccountToken == null) or (.automountServiceAccountToken == true)) | select((.metadata.namespace!="default") and (.metadata.namespace!="kube-system"))' | jq .metadata.namespace | wc -l)
+count_sa=$(kubectl get serviceaccounts --all-namespaces -o json | jq -r '.items[] | select(.metadata.name=="default") | select((.automountServiceAccountToken == null) or (.automountServiceAccountToken == true))' | jq .metadata.namespace | wc -l)
 if [[ ${count_sa} -gt 0 ]]; then
     echo "false"
     exit
 fi
 
-count_rb=$(kubectl get rolebinding --all-namespaces -o json | jq -r '.items[].subjects[] | select(.kind=="ServiceAccount") | select(.name=="default")' | jq .metadata.namespace | wc -l)
-if [[ ${count_rb} -gt 0 ]]; then
-    echo "false"
-    exit
-fi
 
-count_crb=$(kubectl get clusterrolebinding --all-namespaces -o json | jq -r '.items[].subjects[] | select(.kind=="ServiceAccount") | select(.name=="default")' | jq .metadata.namespace | wc -l)
-if [[ ${count_crb} -gt 0 ]]; then
-    echo "false"
-    exit
-fi
+for result in $(kubectl get clusterrolebinding,rolebinding --all-namespaces -o json | jq -r '.items[] | select((.subjects[].kind=="ServiceAccount" and .subjects[].name=="default") or (.subjects[].kind=="Group" and .subjects[].name=="system:serviceaccounts"))' | jq -r '"\(.roleRef.kind),\(.roleRef.name),\(.metadata.namespace)"')
```

Comment:
I think this should be done per namespace, grabbing all can/will be a massive amount of data when dealing with production clusters, doing this also allows the test to fail faster.

Reply:
done